### PR TITLE
Field grouping option for Form class

### PR DESCRIPTION
**Pull request by wilsonge**
WIP at grouping options together using the existing `\Joomla\CMS\Form\Form` class objects

### CODE REVIEW

**Comment by brianteeman** [comment]
As presumably this is going to be applied to all components would it not be better to use/create global strings

**Comment by brianteeman** [comment]
I dont see why this is taking place in the template repo and not the main joomla 4 repo

**Comment by brianteeman** [comment]
> I dont see why this is taking place in the template repo and not the main joomla 4 repo

I clearly forgot about this and just spent an hour doing the same thing on the main repo.

The only problems I see with this PR are
1. I think that the legend should be in the joomla.edit.fieldset (although I havent been able to get the label yet)
2. Your method of looping through the selected fieldsets in an array is OK but that will always result in all the fields being in one column. If I wanted to put the images in the left column and the urls in the right column then that approach wont work

File: administrator/templates/atum/scss/pages/_com_content.scss [hunk 1]
```diff
@@ -27,4 +27,9 @@
 			}
 		}
 	}
+
+	section.options-fieldset {
```
Comment (brianteeman) [inline administrator/templates/atum/scss/pages/_com_content.scss]: We really should be doing the fieldset legend in all the sections so a more generic class will probably be required

Comment (wilsonge) [reply administrator/templates/atum/scss/pages/_com_content.scss]: This is adding a white background on the fieldset. The class isn't to do with the legend

**Comment by wilsonge** [comment]
> I think that the legend should be in the joomla.edit.fieldset (although I havent been able to get the label yet)

This is what we do in `joomla.edit.params` layout i think?

> Your method of looping through the selected fieldsets in an array is OK but that will always result in all the fields being in one column. If I wanted to put the images in the left column and the urls in the right column then that approach wont work

That is true - I've add some basic column count stuff (not very well but I'm sure somebody who is half competent at css could do it better). But to divide it that specific we're probably going to have to render it ourselves rather than in a foreach anyhow

**Comment by brianteeman** [comment]
>> I think that the legend should be in the joomla.edit.fieldset (although I havent been able to get the label yet)

> This is what we do in joomla.edit.params layout i think?

Pretty sure I tried that approach but will try again

**Comment by brianteeman** [comment]
Instead of manually grouping the fieldsets that you want to display in one tab into an array could we not use the `<fields>` element instead.  That way the xml would look like
```
<fields name ="options" label="options">
	<fieldset name="set1" label="set1">
	</fieldset>
	<fieldset name="set2" label="set2">
	</fieldset>
	<fieldset name="set3" label="set3">
	</fieldset>
</fields>
```
This would remove the need to create your `fieldsetsInOptions` variable and probably would also remove the need for the `ignore_fieldsets`. It might also remove the need for the `uitab.addTab` etc as it would all be in the layout. And we would be able to remove some of the layouts as well

**Comment by wilsonge** [comment]
My recollection is that the fields element forces json encoding when saving to the database?

**Comment by brianteeman** [comment]
No idea. I was wondering what it does

**Comment by bembelimen** [comment]
The ```<fields>``` tag creates a hierarchy in field naming like:
```
<fields name="foo">
  <fields name="bar">
    <field name="buz" />
  </fields>
</fields>
```

Leads to (assuming, we use "jform"): ```input type="text" name="jform[foo][bar][buz]" ... >```

There is no auto-JSONfying, but there is a "hardcoded" converter in the table: https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Table/Table.php#L648-L657

**Comment by bembelimen** [comment]
Merged into https://github.com/joomla/backend-template/pull/334

Thx.

**Pull request closed without merging**